src/junitParser.ts

```diff
@@ -70,26 +70,36 @@ async function parseFile(file: string, projectTokenDictionaryStrs: string[]) {
     const data: string = fs.readFileSync(file, 'utf8');
     const parser = new XMLParser({ allowBooleanAttributes: true, ignoreAttributes: false, attributeNamePrefix: '' });
     const report = parser.parse(data) as Partial<JUnitReport>;
-
-    return parseSuite(report, file, projectTokenDictionaryStrs);
+    const testsuites = castArray(report.testsuites?.testsuite);
+    const result: InternalTestResult[] = [];
+    for (const testsuite of testsuites) {
+        await parseSuite(file, projectTokenDictionaryStrs, testsuite as JUnitSuite, result);
+    }
+    return result;
```

Comment:
Why pass the `result` array instead of mapping here and then not having to change `parseSuite`?
```suggestion
    return Promise.all(testsuites.map(testsuite => parseSuite(file, projectTokenDictionaryStrs, testsuite as JUnitSuite));
```

Reply:
When you are right you are right